### PR TITLE
adopt source available license

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Adopt the source available license for this plugin.

#### Ticket Link
Fixes: https://mattermost.atlassian.net/browse/MM-55748

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,7 +10,11 @@
...
+
+Although a valid Mattermost Enterprise Edition E20 license is required if using this plugin in production, the [Mattermost Source Available License](LICENSE) allows you to compile and test this plugin in development and testing environments without a Mattermost Enterprise Edition E20 license. As such, we welcome community contributions to this plugin.
+
+On startup, the plugin checks for a valid Mattermost Enterprise Edition E20 license. If you're running an Enterprise Edition of Mattermost and don't already have a valid license, you can obtain a trial license from **System Console > Edition and License**. If you're running the Team Edition of Mattermost, including when you run the server directly from source, you may instead configure your server to enable both testing (`ServiceSettings.EnableTesting`) and developer mode (`ServiceSettings.EnableDeveloper`). These settings are not recommended in production environments.
```
Comment (lieut-data) [inline README.md:17]:
```suggestion
If you're running an Enterprise Edition of Mattermost and don't already have a valid license, you can obtain a trial license from **System Console > Edition and License**. If you're running the Team Edition of Mattermost, including when you run the server directly from source, you may instead configure your server to enable both testing (`ServiceSettings.EnableTesting`) and developer mode (`ServiceSettings.EnableDeveloper`). These settings are not recommended in production environments.
```

**Review by cwarnermm — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -10,7 +10,11 @@
 
 ## License
 
-See the [LICENSE](./LICENSE) file for license rights and limitations.
+This repository is licensed under the [Mattermost Source Available License](LICENSE) and requires a valid Enterprise Edition E20 license when used for production. See [frequently asked questions](https://docs.mattermost.com/overview/faq.html#mattermost-source-available-license) to learn more.
```
Comment (johndavidlugtu) [inline README.md:13]: "E20 license" switch to "Enterprise license"

File: README.md [hunk 3]
```diff
@@ -10,7 +10,11 @@
...
+
+Although a valid Mattermost Enterprise Edition E20 license is required if using this plugin in production, the [Mattermost Source Available License](LICENSE) allows you to compile and test this plugin in development and testing environments without a Mattermost Enterprise Edition E20 license. As such, we welcome community contributions to this plugin.
```
Comment (johndavidlugtu) [inline README.md:15]: Same here

File: LICENSE [hunk 4]
```diff
@@ -1,201 +1,39 @@
...
-   limitations under the License.
+The Mattermost Source Available License license (the “Source Available License”)
+Copyright (c) 2015-present Mattermost
+
+With regard to the Mattermost Software:
+
+This software and associated documentation files (the "Software") may only be
+used in production, if you (and any entity that you represent) have agreed to,
+and are in compliance with, the Mattermost Terms of Service, available at
+https://mattermost.com/enterprise-edition-terms/ (the “EE Terms”), or other
```
Comment (johndavidlugtu) [inline LICENSE:9]: @cwarnermm isn't this the updated software license term: https://mattermost.com/software-services-license-agreement/

File: LICENSE [hunk 5]
```diff
@@ -1,201 +1,39 @@
...
+agreement governing the use of the Software, as agreed by you and Mattermost,
+and otherwise have a valid Mattermost Enterprise E20 subscription for the
```
Comment (johndavidlugtu) [inline LICENSE:11]: E20 switch to Enterprise

File: LICENSE [hunk 6]
```diff
@@ -1,201 +1,39 @@
...
+correct number of user seats. Subject to the foregoing sentence, you are free
+to modify this Software and publish patches to the Software. You agree that
+Mattermost and/or its licensors (as applicable) retain all right, title and
+interest in and to all such modifications and/or patches, and all such
+modifications and/or patches may only be used, copied, modified, displayed,
+distributed, or otherwise exploited with a valid Mattermost Enterprise E20
```
Comment (johndavidlugtu) [inline LICENSE:17]: E20 -> Enterprise

**Review by johndavidlugtu — COMMENTED** [review]
Just need to change E20 references to Enterprise as well as use the updated Software License Terms

**Comment by lieut-data** [comment]
Thanks, @johndavidlugtu! @cwarnermm, do you recall when we updated the text in https://docs.mattermost.com/about/faq-mattermost-source-available-license.html? 

It looks like it matches what @johndavidlugtu expects, but doesn't match my memory of the language approved by LEGAL at the time this license was drafted.

I've copied these changes from our other repositories using the source available license, so wondering if we need to make a holistic change everywhere.

**Comment by lieut-data** [comment]
@johndavidlugtu, I've updated the LICENSE to reflect the newer language on https://docs.mattermost.com/about/faq-mattermost-source-available-license.html -- we'll need to go back and do this for Playbooks as well.

I've /not/ updated https://mattermost.com/enterprise-edition-terms/ to point at https://mattermost.com/software-services-license-agreement/ as these are different documents at present. Propose we circle back with legal to update the source available license holistically and then push that everywhere as needed. Thoughts?

**Review by johndavidlugtu — APPROVED** [review]
That works. Some context: the reason https://mattermost.com/enterprise-edition-terms/ still exists is primarily due to some existing customers having that as part of their contract. Moving forward though, legal wants all new documents to use https://mattermost.com/software-services-license-agreement/

By the way, I realize "Mattermost Enterprise Edition E20 license" should technically update to "Mattermost Enterprise Edition Enterprise license" since Enterprise refers to both edition (TE vs E0) vs license (Professional, Enterprise).

That said, "Mattermost Enterprise Edition Enterprise license" sounds a bit redundant. I'll also check this with legal.

**Review by fmartingr — APPROVED** [review]

**Pull request merged**